### ISSUE TITLE
Using phoenix without leiningen

### ISSUE THREAD

**Issue by piranha** [james-henderson/phoenix#4]
I'm trying to use phoenix together with [boot](http://boot-clj.com), and there are few problems. First is that `phoenix/init-phoenix!` is private function, but I can just step over it:

```
(require '[phoenix])
(#'phoenix/init-phoenix! {:phoenix/config "myapp.edn"})
```

But then starting it fails with:

```
boot.user=> (phoenix/start!)
:reloading ()

java.lang.ClassCastException: clojure.core$some_fn$sp2__6490 cannot be cast to clojure.lang.ILookup
```

I checked that `phoenix/config-resource` points at existing file and wonder what's the best way to debug this.

**Comment by jarohen** [comment]
Hi Alexander,

I was wondering whether it'd be worth releasing the Phoenix runtime separately - your ticket makes me think it probably is! At the very least, I could make `init-phoenix!` public so that anyone can call it. 

Do you have a stack trace for that error that you could paste somewhere? 

Cheers,

James

**Comment by sansolovyov** [comment]
That's actually all output I've got! And I'm not sure how to get more...

**Comment by jarohen** [comment]
If you're running it in the REPL, could you try `(.printStackTrace *e)` - '*e' is the most recent exception to be thrown in the REPL.

**Comment by sansolovyov** [comment]
Hm, managed to get a bit more:

```
piranha@spica ~/dev/web/mailers> boot repl
:reloading ()
  clojure.lang.ExceptionInfo: clojure.core$some_fn$sp2__6490 cannot be cast to clojure.lang.ILookup
    data: {:file
           "../var/folders/yh/786fldyj33g0g9zh09fqh0dr0000gn/T/boot.user5884189937681505049.clj",
           :line 19}
java.lang.ClassCastException: clojure.core$some_fn$sp2__6490 cannot be cast to clojure.lang.ILookup
                                    ...                                        
               clojure.core/read-string                          core.clj: 3497
            phoenix.config/parse-config                        config.clj:   27
             phoenix.config/load-config                        config.clj:   51
             phoenix.config/read-config                        config.clj:  121
                    phoenix/make-system                       phoenix.clj:   26
                      phoenix/do-start!                       phoenix.clj:   46
                                    ...                                        
clojure.tools.namespace.repl/do-refresh                          repl.clj:   99
   clojure.tools.namespace.repl/refresh                          repl.clj:  142
                                    ...                                        
                         phoenix/start!                       phoenix.clj:   52
                       boot.user/eval98  boot.user5884189937681505049.clj:   21
                                    ...                                        
                     boot.main/-main/fn                          main.clj:  137
                        boot.main/-main                          main.clj:  137
                                    ...                                        
                       boot.App.runBoot                          App.java:  224
                          boot.App.main                          App.java:  316
```

I'll try to understand what is stripping it. And one second, I'll try your suggestion.

**Comment by sansolovyov** [comment]
Oh! Found a problem. :) I've added `:phoenix/includes [#phoenix/file "~/.myapp/passwords.edn"]` so that I will not forget about it and it doesn't exist. Just needs a bit better error message. :)

**Comment by jarohen** [comment]
Ah - yes, Phoenix should have a better error message than that! Actually, I suspect that's a different issue, will have a look now.

**Comment by sansolovyov** [comment]
When I commented this `#phoenix/file` thing out, I've got completely different error - because I forgot one dependency. :)

**Comment by jarohen** [comment]
Just deploying 0.0.2-SNAPSHOT to Clojars with a fix for that issue. Have also made `(phoenix/init-phoenix!)` public, so you should be able to call it as `(phoenix/init-phoenix! (io/resource "config-file.edn"))` (n.b. now passing in an io/resource rather than just a file name)

Let me know if this fixes the above issue? It should handle missing files ok, and write a warning out to the console?

James

**Comment by sansolovyov** [comment]
Looks like that, understandable but I guess still could be improved:

```
:reloading ()
Jan 25, 2015 4:04:33 PM clojure.tools.logging$eval79$fn__83 invoke
WARNING: Can't find config: #<File /Users/piranha/.mailers/passwords.edn>
    clojure.lang.ExceptionInfo: 
    data: {:file
           "../var/folders/yh/786fldyj33g0g9zh09fqh0dr0000gn/T/boot.user216406933021673177.clj",
           :line 19}
java.lang.NullPointerException: 
                                    ...                                       
               clojure.core/read-string                         core.clj: 3497
            phoenix.config/parse-config                       config.clj:   27
             phoenix.config/load-config                       config.clj:   51
             phoenix.config/read-config                       config.clj:  140
                    phoenix/make-system                      phoenix.clj:   26
                      phoenix/do-start!                      phoenix.clj:   46
                                    ...                                       
clojure.tools.namespace.repl/do-refresh                         repl.clj:   99
   clojure.tools.namespace.repl/refresh                         repl.clj:  142
                                    ...                                       
                         phoenix/start!                      phoenix.clj:   52
                       boot.user/eval98  boot.user216406933021673177.clj:   21
                                    ...                                       
                     boot.main/-main/fn                         main.clj:  137
                        boot.main/-main                         main.clj:  137
                                    ...                                       
                       boot.App.runBoot                         App.java:  224
                          boot.App.main                         App.java:  316
```

**Comment by jarohen** [comment]
Sure - what other information would you expect to see here?

**Comment by sansolovyov** [comment]
Well, I just thought there is no need in NPE - but amount of information is certainly enough. :-)

**Comment by jarohen** [comment]
No worries - found and hidden the NPE - should have disappeared in the newest snapshot?

**Comment by jarohen** [comment]
Just released 0.0.2 stable with these changes in. Thanks for your suggestions :)

James

**Comment by sansolovyov** [comment]
Yep, works nicely. :-)
